include/openrave/openravejson.h

```diff
@@ -614,37 +640,68 @@ inline void SaveJsonValue(rapidjson::Value& rTransform, const OpenRAVE::RaveTran
 }
 
 template<class T>
-inline void SaveJsonValue(rapidjson::Value& v, const std::vector<T>& t, rapidjson::Document::AllocatorType& alloc) {
+inline void SaveJsonValue(rapidjson::Value& v, const OpenRAVE::RaveVector<T>& t, rapidjson::Document::AllocatorType& alloc) {
     v.SetArray();
-    v.Reserve(t.size(), alloc);
-    for (size_t ivec = 0; ivec < t.size(); ++ivec) {
+    // TODO, what's a better way to serialize?
+    bool bHas4Values = t[3] != 0;
+    int numvalues = bHas4Values ? 4 : 3;
+    v.Reserve(numvalues, alloc);
+    for (int ivalue = 0; ivalue < numvalues; ++ivalue) {
         rapidjson::Value tmpv;
-        SaveJsonValue(tmpv, t[ivec], alloc);
+        SaveJsonValue(tmpv, t[ivalue], alloc);
         v.PushBack(tmpv, alloc);
     }
 }
 
 template<class T>
-inline void SaveJsonValue(rapidjson::Value& v, const std::vector<T>& t, rapidjson::Document::AllocatorType& alloc, size_t n) {
+inline void SaveJsonValue(rapidjson::Value& v, const OpenRAVE::geometry::RaveOrientedBox<T>& t, rapidjson::Document::AllocatorType& alloc) {
+    v.SetObject();
+    rapidjson::Value tmpv;
+    SaveJsonValue(tmpv, t.extents, alloc);
+    v.AddMember("extents", tmpv, alloc);
+    SaveJsonValue(tmpv, t.transform, alloc);
+    v.AddMember("transform", tmpv, alloc);
+}
+
+inline void SaveJsonValue(rapidjson::Value &rTriMesh, const OpenRAVE::TriMesh& t, rapidjson::Document::AllocatorType& alloc) {
+    rTriMesh.SetObject();
+    rapidjson::Value rVertices;
+    rVertices.SetArray();
+    rVertices.Reserve(t.vertices.size()*3, alloc);
+    for(size_t ivertex = 0; ivertex < t.vertices.size(); ++ivertex) {
+        rVertices.PushBack(t.vertices[ivertex][0], alloc);
+        rVertices.PushBack(t.vertices[ivertex][1], alloc);
+        rVertices.PushBack(t.vertices[ivertex][2], alloc);
+    }
+    rTriMesh.AddMember("vertices", rVertices, alloc);
+
+    rapidjson::Value rIndices;
```

Comment:
why not just `SetJsonValueByKey(rTriMesh, "indices", t.indices, alloc);`

Reply:
same reason with https://github.com/rdiankov/openrave/pull/1213#discussion_r1148720953